### PR TITLE
Add member hub link if user is authenticated

### PR DESCRIPTION
**Pull request by arathjz**
**What:**
Add member hub link if the user is authenticated

**Why:**
Relates to: https://app.asana.com/0/1159164196409156/1194956796156396/f

**How:**
- Include a prop `memberhuburl` as a reference for the member hub link
- Include `Member hub` link if the user is authenticated

#### Media:
image

### CODE REVIEW

File: packages/header-component/src/components/dc-header/dc-header.tsx [hunk 1]
```diff
@@ -94,6 +99,16 @@ export class Header {
     this._links = JSON.parse(newValue);
   }
 
+  @Watch("user")
+  userDidChage(newUser) {
```
Comment (duranmla) [inline packages/header-component/src/components/dc-header/dc-header.tsx:103]: As two objects are going to be different `{} === {} // false` wouldn't this lead to the same issue we used to have?.
While may still have some edge cases, first thing that comes to my mind would be to compare the users' id.

**On top of that,** it feels awkward to mutate the prop like that, **what about something on the render instead?** It is not as sexy, but the main goal is not to mutate the prop value.

Perhaps on https://github.com/debtcollective/packages/blob/b520e1e1d610900ad2d52607bb5cba9773cd7318/packages/header-component/src/components/dc-header/dc-header.tsx#L190 you can do like `{ user.id ? <authLinks> : null }`

File: packages/header-component/src/components/dc-header/dc-header.tsx [hunk 2]
```diff
@@ -94,6 +99,16 @@ export class Header {
...
+    if (newUser) {
```
Comment (duranmla) [inline packages/header-component/src/components/dc-header/dc-header.tsx:104]:
```suggestion
    if (newUser.id !== this.user.id) {
```

**Review by duranmla — CHANGES_REQUESTED** [review]
I'm a bit hesitating with the prop mutation. Maybe there is some doc I'm missing or simple logic I don't know? I guess could be that every time that we execute `userDidChage` there is a fresh reference of the `_links` value? **Are we sure about it?**

**Comment by arathjz** [comment]
@duranmla thanks for the feedback, I've addressed your comments in a new commit

File: packages/header-component/src/components/dc-header/dc-header.tsx [hunk 3]
```diff
@@ -131,6 +121,14 @@ export class Header {
 
   render() {
     const user = this.user;
+    const authLinks = [
+      {
+      text: 'Member hub',
```
Comment (duranmla) [inline packages/header-component/src/components/dc-header/dc-header.tsx:126]: **Nit.** have proper indentation.
```
{
    text:
}
```
instead
```
{
text
}

File: packages/header-component/src/components/dc-header/dc-header.tsx [hunk 4]
```diff
@@ -131,6 +121,14 @@ export class Header {
 
   render() {
     const user = this.user;
+    const authLinks = [
```
Comment (duranmla) [inline packages/header-component/src/components/dc-header/dc-header.tsx:124]: **Nit.** this array is not just `authLinks`, maybe:

```
const authLinks = [/* member link */]

render -> 
  const linksToRender = this.user?.id ? [...authLinks, ...this._links] : this._links;
```

**Review by duranmla — APPROVED** [review]
Couple minor things but I feel we can address and then merge.

**Pull request merged**